[openimageio] Update to v3.0.1.0

donny-dont opened the pull request:
- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [X] SHA512s are updated for each updated download.
- [ ] ~The "supports" clause reflects platforms that may be fixed by this new version.~
- [ ] ~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~
- [ ] ~Any patches that are no longer applied are deleted from the port's directory.~
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is added to each modified port's versions file.

dg0yt commented on ports/openimageio/portfile.cmake:
```diff
@@ -1,15 +1,22 @@
+vcpkg_download_distfile(FIX_LIBRAW_BUILD_PATCH
+    URLS https://github.com/AcademySoftwareFoundation/OpenImageIO/commit/904df59ab74d0c89c1c9eea7d5ef0ecfe0620b2c.diff?full_index=1
+    FILENAME Fix-libraw-build.patch
```
Nitpick:
```suggestion
    FILENAME AcademySoftwareFoundation-OpenImageIO-libraw-build.patch
```
to group files in `downloads/`.

donny-dont replied:
Is there any reason why there isn’t a vcpkg_patch_from_github function that codifies this pattern @dg0yt ? Was thinking about adding it.

dg0yt replied:
Only few people care about names in `downloads/`.

TBH I would probably want to be able to directly add 
~~~
  https://this.patch:SHA512=abc
~~~
or similar to the `PATCHES` list.

dg0yt reviewed (commented):
AFAICS fix-dependencies.patch and fix-openimageio_include_dir.patch have no relevant changes and could remain unchanged.

donny-dont commented:
> AFAICS fix-dependencies.patch and fix-openimageio_include_dir.patch have no relevant changes and could remain unchanged.

So going forward I should not commit line number changes? Wasn’t sure how good the patching was when the line numbers change.

donny-dont committed:
[openimageio] Update to v3.0.1.0

Recreate the required patches. Attach a patch for fix a libraw build issue found when updating lcms.

Add a feature definition for jpegxl support using libjxl.

Co-authored-by: Kai Pastor <dg0yt@darc.de>

donny-dont committed:
[openimageio] Update to v3.0.1.0

donny-dont committed:
CI

donny-dont commented:
Alrighty should be fixed @Cheney-W @dg0yt  there just appeared to be a hiccup in the CI so pushed an empty commit

BillyONeal commented:
Thanks for the update!

Cheney-W commented:
Feature test passed with below triplets:
x86-windows
x64-windows
x64-windwos-static
